monitoring-6.x.md

```diff
@@ -0,0 +1,948 @@
+
+*   `api_http_requests_total` - differentiate request types: `operation="create|update|delete"`
+
+When troubleshooting problems with metrics, it is sometimes useful to look into the specified container logs where it can be seen if it experiences communication issues with Prometheus service or has other issues:
+
+```
+$ kubectl -nmonitoring logs prometheus-adapter-6586cf7b4f-hmwkf
+```
+
+```
+$ kubectl -nmonitoring logs kube-state-metrics-69594c468-wcr4g kube-state-metrics
+```
+
+```
+$ kubectl -nmonitoring logs node-exporter-hz972 node-exporter
+```
+
+In addition, any other apps that collect metrics should also submit them into the same DB in order for proper retention policies to be enforced. 
+
+## Exploring Prometheus
+
+Like mentioned above, Prometheus is exposed via a cluster-local Kubernetes service `prometheus-k8s.monitoring.svc.cluster.local:9090` and serves its HTTP API on port `9090` so we can use it to explore the database from the CLI.
+
+Also, as seen above we have the following Prometheus pods:
+
+```
+prometheus-adapter-6586cf7b4f-hmwkf   
+prometheus-k8s-0                       
+prometheus-operator-7bd7d57788-mf8xn   
+```
+Prometheus operator for kuebrnetes allows easy monitoring defintions for kubernetes services and deployment and managment of Prometheus instances. 
+
+Prometheus adapter is an API extension for kubernetes that users prometheus queries to populate kubernetes resources and custom metrics APIs.
+
+Let's enter the Gravity master container to make sure the services are resolvable and to get access to additional CLI tools:
+
+```bash
+$ sudo gravity shell
+```
+
+Let's ping the database to make sure it's up and running:
+
+```bash
+$ curl -sl http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/status/config
+// Should return "status":"success" within currently loaded configuration file.
+```
+
+A list of alerting and recording rules that are currently loaded is available by executing:
+
+```bash
+$ curl http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/rules | jq
+```
+Also we can see all metric points, by executing the following command:
+
+```bash
+$ curl http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/query?query=up | jq
+```
+
+Finally, we can query Prometheus using it's SQL-like query language (PromQL) to for example evaluate metrics identified under the expression `up` at the specified time:
+
+```bash
+$ curl 'http://prometheus-k8s.monitoring.svc.cluster.local:9090/api/v1/query?query=up&time=2020-03-13T20:10:51.781Z' | jq
+```
+
+Refer to the Prometheus [API documentation](https://prometheus.io/docs/prometheus/latest/querying/basics/) if you want to learn more about querying the database.
+
+## Metric Retention Policy 
+
+### Time based retention
+
+Users should start Prometheus with the following flag with their desired time.
+
+`--storage.tsdb.retention.time`: Determines when to remove old data. Defaults to 15 days.
```

Comment:
Like above, not sure we should be showing CLI flags as we're configuring Prometheus ourselves and this is not exposed to the user.

What we should mention, however, is our retention period. It's 30 days IIRC. cc @UnderGreen right?

Reply:
Updated to 30 days